lib/cache/cacheManager/factory.ts

```diff
@@ -0,0 +1,27 @@
+import { CacheOptions } from '../CacheOptions';
+import { CacheManager } from './CacheManager';
+import { ClassCacheManager } from './ClassCacheManager';
+import { InstanceCacheManager } from './InstanceCacheManager';
+
+const cacheManagerFactories: ReadonlyMap<
+  'class' | 'instance',
+  <K>(timeout: number, options: CacheOptions) => CacheManager<K>
+> = new Map<'class' | 'instance', <K>(timeout: number, options: CacheOptions) => CacheManager<K>>()
+  .set('class', (timeout, options) => new ClassCacheManager<any>(timeout, options))
+  .set('instance', (timeout, options) => new InstanceCacheManager<any>(timeout, options));
```

Comment:
To be honest this is hard to read. Do you need this map at all? Maybe plain vanilla `if/else` or `switch` ?